### ISSUE TITLE
v14.2.0 adds new required `getIncrement` method breaking existing plugins

### ISSUE THREAD

**Issue by rwjblue** [release-it/release-it#713]
Existing plugins generally depend on `release-it` itself in order to extend from the built-in `Plugin` base class. Unfortunately, that means (due to lock files with both yarn and npm) that often times **multiple** versions of `release-it`'s `Plugin` base class.

<details>
<summary>For example, a sample project using `release-it` and `release-it-lerna-changelog` has these versions</summary>

```
at 15:03:07 ❯ y why release-it
yarn why v1.22.10
[1/4] 🤔  Why do we have the module "release-it"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "release-it@14.2.0"
info Has been hoisted to "release-it"
info This module exists because it's specified in "devDependencies".
info Disk size without dependencies: "848KB"
info Disk size with unique dependencies: "9.2MB"
info Disk size with transitive dependencies: "41.11MB"
info Number of shared dependencies: 124
=> Found "release-it-lerna-changelog#release-it@13.7.1"
info This module exists because "release-it-lerna-changelog" depends on it.
info Disk size without dependencies: "412KB"
info Disk size with unique dependencies: "8.86MB"
info Disk size with transitive dependencies: "40.96MB"
info Number of shared dependencies: 124
✨  Done in 0.81s.
```

</details>

As a result, the change in https://github.com/release-it/release-it/pull/710 is _basically_ a breaking change for users of plugins.

**Comment by rwjblue** [comment]
I think _personally_ that I'd like to resolve this by making `release-it` a `peerDependencies` of my various `release-it` plugins (since we are very confident that the host has to actually provide `release-it` itself), but I'd love to see official guidance here and to surface the general concern.

**Comment by webpro** [comment]
Yes, `release-it` should be a `peerDependency` for plugins.

It's in https://github.com/release-it/plugin-starterkit/blob/master/package.json, but I didn't really think ahead of the issue you faced and document/announce this properly. Apologies for any issues you may have had! Will try to do better next time.

**Comment by rwjblue** [comment]
> It's in https://github.com/release-it/plugin-starterkit/blob/master/package.json,

Gotcha, yeah. Though the starter kit only recently moved `release-it` into `peerDependencies` (in https://github.com/release-it/plugin-starterkit/commit/005713b957ce494e76ea52e99e2ff6ef334ddf9b) and I wasn't watching that repo for changes / releases so I didn't notice it.

> I didn't really think ahead of the issue you faced (i.e. for existing plugins) and document/announce this properly

I think it may make sense to explicitly call this out in the plugin documentation for new plugin authors.

> Apologies for any issues you may have had! Will try to do better next time.

Absolutely no worries here! I mostly just wanted to raise the issue so folks using my release setup have a chance to find the issue when they search github for it 😄.

I'll cross link here when I've got new versions of `release-it-yarn-workspaces` and `release-it-lerna-changelog` published that work properly with v14.2.0.

**Comment by webpro** [comment]
> ...and I wasn't watching that repo for changes / releases so I didn't notice it.

Didn't expect you to! Just pointing that new plugin authors hopefully see it. It wasn't even linked to from the plugin docs, so now it is.

> I think it may make sense to explicitly call this out in the plugin documentation for new plugin authors.

Yes! Just did. Thank you very much for the suggestion.

Thanks again, Robert! Closing this issue for now.
